chrome/content/zotero/xpcom/utilities.js

```diff
@@ -179,9 +179,13 @@ Zotero.Utilities = {
 				var lastName = author;
 			}
 		} else {
-			var spaceIndex = author.lastIndexOf(" ");
-			var lastName = author.substring(spaceIndex+1);
-			var firstName = author.substring(0, spaceIndex);
+			// Don't parse "Firstname Lastname [Country]" as "[Country], Firstname Lastname"
+			var spaceIndex = author.length;
+			do {
+				spaceIndex = author.lastIndexOf(" ", spaceIndex-1);
+				var lastName = author.substring(spaceIndex + 1);
+				var firstName = author.substring(0, spaceIndex);
+			} while (!/\w/.test(lastName[0]) && spaceIndex > 0)
```

Comment:
> `\w` and the like aren't really usable for user input (as surely you would agree)

I do agree. I was looking for a better solution that includes extended characters, but couldn't find anything obvious online and there is also code just above, which has a regexp

```
var allCaps = 'A-Z' + 
 	'\u0400-\u042f';		//cyrilic
```

that made go for a less-than-ideal solution.